src/mechanics/cell/SimpleAssembler.h

```diff
@@ -20,6 +20,17 @@ class SimpleAssembler
     GlobalDofMatrixSparse BuildMatrix(const Group<CellInterface>& cells, std::vector<DofType> dofTypes,
                                       CellInterface::MatrixFunction f) const;
 
+    //! @brief Assembles a diagonally lumped matrix from local matrices calculated by f
+    //! @param cells group of cells to be used for assembly
+    //! @param dofTypes vector of dofTypes
+    //! @return a global dof vector that represents a collection of diagonal matrices
+    //! @remark HRZ lumping is used here, the assumptions made are:
+    //! - shape functions sum to 1, only then the total mass for a cell can be calculated by
+    //!   summing over all entries of the local mass matrix
+    //! - the total mass of a cell is the same for all components of the considered dof
```

Comment:
Do you think it is required to somehow check these assumptions?

Reply:
I don't want to and leave it to the user and the doc.
1) Partition of unity (first assumption) is also essential elsewhere and this might not be the ideal place to test for it
2) Currently vector valued (multicomponent) dofs use the same interpolation for each component. If this changes one has to rethink the algorithms at several places.
-> so these checks might be included later